azure-core/azure/core/exceptions.py

```diff
@@ -73,48 +76,49 @@ def raise_with_traceback(self):
             self.__traceback__ = self.exc_traceback
             raise self
 
-
-class AzureLibraryError(AzureError):
-    """An error occurred in the client while processing the pipeline."""
-
-
-class AzureLibraryResponse(AzureLibraryError):
-    """An error occurred in the client while processing the response."""
-
-
-class AzureLibraryRequest(AzureLibraryError):
-    """An error occurred in the client while building the request."""
-
-
-class DecodeError(AzureLibraryResponse):
+class DecodeError(AzureError):
     """Error raised during response deserialization."""
 
-
 class ServiceRequestError(AzureError):
     """An error occurred while attempt to make a request to the service."""
 
 
-class ConnectionError(ServiceRequestError):
+class ConnectError(ServiceRequestError):
     """An error occurred while attempting to establish the connection.
     These errors are safe to retry."""
 
 
-class ConnectionTimeoutError(ConnectionError):
-    """The request timed out while trying to connect to the remote server.
-    These errors are safe to retry."""
+class ReadTimeoutError(ServiceRequestError):
+    """The server did not send any data in the allotted amount of time.
+    These errors may not be safe to retry."""
 
 
-class ConnectionReadError(ServiceRequestError):
-    """An error occurred during the request/response.
-    These errors may not be safe to retry."""
+class ServiceResponseError(AzureError):
```

Comment:
Nothing seems to use ServiceResponseError at the moment? If this is for a request the has reached the service then should it not be of the same family as ReadTimeoutError?

Reply:
Yes. Yes, it should.

Reply:
Okay - so your intention would be to make ReadTimeoutError of type ServiceResponseError?